cores/esp8266/uart.c

```diff
@@ -72,6 +72,18 @@ struct uart_
     struct uart_rx_buffer_ * rx_buffer;
 };
 
+struct uartIsrContext_t
+{
+    uartInterruptHandler callback;
+    void* arg;
+};
+
+// NOTE: GCC will generated an invalid warning for the following line
+// see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53119
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-braces"
+static volatile struct uartIsrContext_t s_uartInterruptContext[2] = { 0 };
```

Comment:
You can remove all these pragmas by simply dropping `{0}`.  It's a global static, so is 0-initted before the app begins.

Reply:
They are not guaranteed to be initialized, it based on the flags used to compile.  So it wouldn't be safe to do that.
Consensus from my searches is to use the pragma until/if it gets fixed.

Reply:
It's guaranteed on any working C compiler (this is a *.c file).  There might be something weird in C++, but that's not material in this file.  The only concern would be if your code comes before the main C startup routines, (i.e. in early_preinit() callbacks/etc.).  You can verify by looking at the .MAP file and see that it's in the bss segment.

Reply:
@earlephilhower is right here — you can drop the initializer.

Reply:
fixed